### ISSUE TITLE
Loading from google cloud storage using a recursive wildcard stopped working.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [runt18/google-bigquery#190]
```
We were loading bigquery tables using the following command:
.python-env/bin/bq load --source_format=CSV --allow_quoted_newlines=1 
--field_delimiter=tab --quote= --replace=1 --replace=1 --job_id <job-number> 
<schema.table> gs://some/path/** google/bigquery/schemas/raw/table.json

Recently it stopped working with the following error given in bq show -j 
<job-number>:
"Errors encountered during job execution. Not found: URI gs://some/path/**"
Unfortunately our log was overwritten so we can not check when it exactly 
started happening.

I'm using "This is BigQuery CLI v2.0.17" on a mac and on ubuntu 10.04
```

Original issue reported on code.google.com by `tho...@pagerduty.com` on 13 Nov 2014 at 5:33

**Comment by GoogleCodeExporter** [comment]
```
As a workaround, can you please try a single wildcard character, '*'? This 
should have a "recursive" effect.

We recently had a change for improved single-character wildcard support. The 
recursive wildcard you referenced, '**', is a gsutil client semantic: 
https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames#directory-by
-directory-vs-recursive-wildcards.

The gsutil client provides a hierarchical file tree illusion. In contrast, the 
Cloud Storage API treats a path like "gs://bucket/abc/def/ghi.txt" as an object 
with the '/' character in the object name (where the object name is 
"abc/def/ghi.txt"). See 
https://cloud.google.com/storage/docs/gsutil/addlhelp/HowSubdirectoriesWork.

Although we did not intend to support '**' in the bq client initially, we can 
consider it. Apologies for the break!

Furthermore, we have a separate internal bug to improve the documentation. 
Thank you for identifying this issue!
```

Original comment by `bch...@google.com` on 13 Nov 2014 at 7:40

**Comment by GoogleCodeExporter** [comment]
Original comment by `bch...@google.com` on 13 Nov 2014 at 7:41
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
'*' worked, thanks a lot for the quick response!
```

Original comment by `tho...@pagerduty.com` on 13 Nov 2014 at 9:36

**Comment by GoogleCodeExporter** [comment]
```
Looks like this was resolved as of #3.
```

Original comment by `jcon...@google.com` on 1 Mar 2016 at 12:22
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
bq load --allow_jagged_rows -F tab --quote="" 
<project-name>:<dataset-name>.<table-name>  gs://some/path/* <schema>
Waiting on bqjob_r51ee37f56558cb4_0000015331e95c36_1 ... (0s) Current status: 
DONE
BigQuery error in load operation: Error processing job 
'<project-name>:bqjob_r51ee37f56558cb4_0000015331e95c36_1': Not found: Uri
gs://some/path/mon=2015-02                

from 26th Feb onward '*' not working for us  , am using  BigQuery CLI 2.0.22 on 
AWS linux 
```

Original comment by `aakum...@gmail.com` on 1 Mar 2016 at 1:44

**Comment by GoogleCodeExporter** [comment]
```
This bug was a separate issue.

You've hit a known issue in our import pipeline, and if you send us your 
project ID or project number we'll see if we can address it for you. You can 
also email directly to jcondit@google.com if you'd prefer not to post your 
project number.
```

Original comment by `jcon...@google.com` on 1 Mar 2016 at 6:05

**Comment by GoogleCodeExporter** [comment]
```
Thanks ... 

Project ID
bdp-data-415
Project number
614740166533
```

Original comment by `aakum...@gmail.com` on 2 Mar 2016 at 12:53

**Comment by GoogleCodeExporter** [comment]
```
Thanks. We'll apply a config change to your project within the next couple 
hours that should address the problem.

In the future, I'd recommend filing new bugs for new issues to make sure they 
get tracked properly. If it's the same as an existing issue, we'll merge it for 
you.
```

Original comment by `jcon...@google.com` on 2 Mar 2016 at 9:54

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the support .. is it a problem of our environment , setup or config? 
wonder how to avoid it in next time if it from our side and coz do not know the 
what caused this... anyway thanks for the help.. 
```

Original comment by `aakum...@gmail.com` on 3 Mar 2016 at 11:12

**Comment by GoogleCodeExporter** [comment]
```
It's a bug on our side caused by a change in how our imports work. Sorry about 
the trouble! There are no changes you need to make, and feel free to file bugs 
here if you see other issues in the future.
```

Original comment by `jcon...@google.com` on 3 Mar 2016 at 6:04
